pkg/config/validation/validation.go

```diff
@@ -735,8 +735,8 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 			return nil, err
 		}
 
-		if len(rule.Egress) == 0 {
-			return nil, fmt.Errorf("sidecar: missing egress")
+		if len(rule.Egress) == 0 && len(rule.Ingress) == 0 && rule.OutboundTrafficPolicy == nil {
```

Comment:
can we doc the default egress is */*, and allow empty settings. So that we donot need to update here once we need to add another field

Reply:
Sorry, not sure why we would want to allow an empty config?

Reply:
it is valid according to the doc

	// Egress specifies the configuration of the sidecar for processing
	// outbound traffic from the attached workload instance to other
	// services in the mesh. If not specified, inherits the system
	// detected defaults from the namespace-wide or the global default Sidecar.

Reply:
Why would you make a sidecar like:
```
apiVersion: networking.istio.io/v1alpha3
kind: Sidecar
metadata:
  name: sidecar
spec: {}
```
though? I guess it overrides the root namespace one, so maybe it is useful to allow this?

Reply:
Allow .spec empty is not a good practice. So i agree with the current fix

Reply:
Then does this look good? or anything I need to change?

Reply:
lgtm, let's move forward

Reply:
rebased